spec/features/work_packages/is_empty_filter_spec.rb

```diff
@@ -0,0 +1,111 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require 'spec_helper'
+
+RSpec.describe 'Filter by date with "is empty"', js: true do
+  let(:user) { create(:admin) }
+  let(:project) { create(:project) }
+
+  let(:wp_table) { Pages::WorkPackagesTable.new(project) }
+  let(:filters) { Components::WorkPackages::Filters.new }
+
+  let(:member) do
+    create(:member,
+           user:,
+           project:,
+           roles: [create(:role)])
+  end
+
+  let(:work_package_with_start_date) do
+    create(:work_package,
+           project:,
+           start_date: Time.zone.today)
+  end
+
+  let(:work_package_without_dates) do
+    create(:work_package,
+           project:)
+  end
+
+  let(:work_package_with_custom_date) do
+    create(:work_package,
+           project:)
+  end
+
+  let(:custom_field_date) do
+    create(:date_wp_custom_field,
+           name: 'Date CF',
+           is_filter: true,
+           searchable: true).tap do |custom_field|
```

Comment:
Recently learned that FactoryBot yields the object to the factory if a block is provided; no need to `#tap` :)
```suggestion
           searchable: true) do |custom_field|
```